code/modules/modular_computers/file_system/programs/engineering/shutoff_valve.dm

```diff
@@ -0,0 +1,61 @@
+/datum/computer_file/program/shutoff_valve
+	filename = "shutoffcontrol"
+	filedesc = "Automatic Shutoff Valve Monitoring"
+	nanomodule_path = /datum/nano_module/program/shutoff_monitor
+	program_icon_state = "atmos_control"
+	program_key_state = "atmos_key"
+	program_menu_icon = "shuffle"
+	extended_desc = "This program allows remote control and monitoring of shutoff valves."
+	required_access = list(access_atmospherics)
+	requires_network = 1
+	network_destination = "atmospheric control system"
+	requires_network_feature = NETWORK_SYSTEMCONTROL
+	category = PROG_ENG
+	size = 10
+
+/datum/nano_module/program/shutoff_monitor
+	name = "Shutoff valve monitor"
+
+/datum/nano_module/program/shutoff_monitor/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1, var/datum/topic_state/state = GLOB.default_state)
+	var/list/data = host.initial_data()
+	data["valves"] = list()
+	for(var/obj/machinery/atmospherics/valve/shutoff/S in shutoff_valves)
+		data["valves"][++data["valves"].len] = list("name" = S.name, "enable" = S.close_on_leaks, "open" = S.open, "location" = get_area(S), "ref" = "\ref[S]")
```

Comment:
There are sometimes cosmetic issues with using area rather than area.name in html; unsure if relevant here.